src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -176,164 +127,171 @@ const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
+      // Show tableIcon here as part of the link.
+      return (
+        <Box key={artifactId}>
+          <Link
+            to={`${getPathPrefix()}/workflow/${workflowId}/result/${workflowDagResultId}/artifact/${artifactId}`}
+            component={RouterLink as any}
+            sx={{ marginLeft: '16px' }}
+            underline="none"
+          >
+            {artifactResult.name}
+          </Link>
+        </Box>
+      );
+    }
 
-      return null;
-    });
-  };
+    return (
+      <Box key={artifactId}>
+        <Typography variant="body1">
+          {artifactResult.result.content_serialized}
+        </Typography>
+      </Box>
+    );
+  });
 
   // TODO: refactor getOperatorInput and getOperatorOutput to just one function.
   // was playing around with design before i figured they can be the same.
-  const getOperatorInput = () => {
-    if (!operator || !operator.inputs) {
-      return <CircularProgress />;
+  const operatorInputsList = operator.inputs.map((artifactId, index) => {
+    const artifactResult = (workflowDagResultWithLoadingStatus.result
+      ?.artifacts ?? {})[artifactId];
+    if (!artifactResult) {
+      return null;
     }
 
-    return operator.inputs.map((artifactId, index) => {
-      const artifactResult = workflow.artifactResults[artifactId];
-      if (!artifactResult) {
-        return null;
-      }
-
-      if (artifactResult.result) {
-        return (
-          <ListItem divider key={`metric-input-${index}`}>
-            <Box display="flex">
-              <Box
-                sx={{
-                  width: '16px',
-                  height: '16px',
-                  color: 'rgba(0,0,0,0.54)',
-                }}
-              >
-                <FontAwesomeIcon
-                  icon={
-                    artifactTypeToIconMapping[
-                      artifactResult.result.artifact_type
-                    ]
-                  }
-                />
-              </Box>
-              <Link
-                to={`${getPathPrefix()}/workflow/${workflowId}/result/${workflowDagResultId}/artifact/${artifactId}`}
-                component={RouterLink as any}
-                sx={{ marginLeft: '16px' }}
-                underline="none"
-              >
-                {artifactResult.result.name}
-              </Link>
-            </Box>
-          </ListItem>
-        );
-      }
-
-      return null;
-    });
-  };
+    return (
+      <ListItem divider key={`metric-input-${index}`}>
+        <Box display="flex">
+          <Box
+            sx={{
+              width: '16px',
+              height: '16px',
+              color: 'rgba(0,0,0,0.54)',
+            }}
+          >
+            <FontAwesomeIcon
+              icon={artifactTypeToIconMapping[artifactResult.type]}
+            />
+          </Box>
+          <Link
+            to={`${getPathPrefix()}/workflow/${workflowId}/result/${workflowDagResultId}/artifact/${artifactId}`}
+            component={RouterLink as any}
```

Comment:
Why do we need "as any"?

Reply:
Defer this to @agiron123 who added the original version. I assume it's to bypass some type checking